core/consensus/xpoa/xpoa.go

```diff
@@ -250,24 +251,17 @@ func (xpoa *XPoa) initBFT(cfg *config.NodeConfig, cryptoClient crypto_base.Crypt
 
 	// initialize bft
 	bridge := bft.NewDefaultCbftBridge(xpoa.bcname, xpoa.ledger, xpoa.lg, xpoa)
-	qcNeeded := 3
-	qc := make([]*pb.QuorumCert, qcNeeded)
+	qc := make([]*pb.QuorumCert, 3) // 3为chained-bft的qc存储数目
 	meta := xpoa.ledger.GetMeta()
 	if meta.TrunkHeight != 0 {
 		blockid := meta.TipBlockid
-		for qcNeeded > 0 {
-			qcNeeded--
-			block, err := xpoa.ledger.QueryBlock(blockid)
-			if err != nil {
-				xpoa.lg.Warn("initBFT: get block failed", "error", err, "blockid", string(blockid))
-				return err
-			}
-			qc[qcNeeded] = block.GetJustify()
-			blockid = block.GetPreHash()
-			if blockid == nil {
-				break
-			}
+		block, _ := xpoa.ledger.QueryBlock(blockid)
```

Comment:
QueryBlock is a read disk query operation, there is a risk of failure.It is suggested to deal with errors and prevent panic.